### ISSUE TITLE
Metrics screen gives no feedback that mining is suspended while updating chain

### ISSUE THREAD

**Issue by daira** [zcash/zcash#2139]
When the metrics screen and mining are enabled, but the chain is being updated just after starting zcashd, the following is displayed:

> - You have contributed 0.0000 Sol/s on average to the network solution rate.
> - You have completed 0 Equihash solver runs.

This state can persist for a long time if the zcashd instance has a lot to catch up on (for example for a new instance).

The metrics screen should give feedback that the chain is still being updated and therefore mining has not started yet.

**Comment by daira** [comment]
This also causes the Sol/s rate to be low initially when the node does start mining; perhaps it should only measure the rate from when mining actually starts.

**Comment by str4d** [comment]
Agreed on both points. I initially implemented the simplest-possible screen in order to have it ready for 1.0.0, which just measured `num_solutions_found/time_since_start` (see the `MarkStartTime()` call in `init.cpp`). A better metric would only be measuring time while the miner was actually running, but that would either require a more complex time-sum (to account for times when the miner is paused, either via RPC or when creating a JoinSplit), or an actual timer-like class that can be started, paused and stopped (instead of relying on `GetTime()`).

**Comment by daira** [comment]
+1 for abstracting out a timer class that can measure the rate of events when an event-generating thing is enabled.

**Comment by str4d** [comment]
I started implementing this, and came across a design question: should the local Sol/s be the average since the miner started, or the average miner speed? The two are subtly different, and I chose the current wording specifically to imply the former.

- If we want the former, then we just want to start counting when the miner first starts, and continue regardless of any pausing (so the metric *is* the average Sol/s contributed to the network).
- If we want the latter, then we sum up each mining period, and thus ignore the effects of miner downtime while creating JoinSplits.
